docs/tutorial.md

```diff
@@ -0,0 +1,656 @@
+
+This means that template authors can't blindly follow relationships and incur extra queries, because the attributes that express those relationships simply don't exist in the template context unless they are explicitly fetched. But to mitigate the burden of having to fine-tune our querysets "by hand" for efficiency, we use a high-level "spec" to describe the fields and relationships we need, and `django-readers` is responsible for building the queryset and converting the model instances into dictionaries.
+
+## Requirement 1 revisited: a list of book titles (with `django-readers`)
+
+Let's go back to our initial requirement: just show a list of book titles. Remember that we only needed the `title` field, but Django wastefully extracted _all_ of the fields on the `Book` model from the database. `django-readers` fixes this:
+
+```python
+from django_readers import specs
+
+
+def book_list(request):
+    spec = [
+        "title",
+    ]
+    prepare, project = specs.process(spec)
+    queryset = prepare(Book.objects.all())
+    return render(
+        "book_list.html",
+        {"books": [project(book) for book in queryset]},
+    )
+```
+
+As you can see, at its simplest a `django-readers` spec is just a list of field names. `specs.process` takes the spec and returns _a pair of functions_, which (by convention) we call `prepare` and `project`.
+
+!!! note
+    `prepare` and `project` are `django-readers`-specific terms. There is one other `django-readers` concept beginning with P, `produce`, which will be introduced fully later, but here are their rough definitions to set the scene:
+
+    A `prepare` function takes a queryset and returns another queryset with one or more modifications applied (like `prefetch_related` or `only`: just like chaining queryset methods). This process is called "preparing the queryset".
+
+    A `produce` function (a "producer") takes a model instance and returns ("produces") a value derived from that instance. Often this is the value of an attribute on the model.
+
+    A `project` function (a "projector") takes a model instance and returns a dictionary (the "projection") which maps one or more names (the dictionary keys) to one or more values derived from the instance (often the values of one or more attributes and possibly nested dictionaries representing related models). It _projects_ your data layer into your application's business logic domain. Think of the dictionary returned by a projector (the "projection") as the simplest possible domain object. Generally speaking, it's not necessary to write your own projector functions: you can wrap a producer function to bind the value it returns to a name.
```

Comment:
Assuming the diataxis quadrant and this is for tutorial/quickstart, i think this entire page should start with the following

## Retrieve 1 instance

in views.py
```python

from django_readers import specs


def book_details(request, pk):
    spec = [
        "title",
    ]
    prepare, project = specs.process(spec)
    queryset = prepare(Book.objects.all()).get(pk=pk)
	book = project(queryset)
    return render(
        "book_details.html",
        {"book": book},
    )
```

## Retrieve a list of many instances

this is in your intro but no harm repeating

Reply:
I think we've never hit that gotcha because we don't tend to use `__str__` much. Usually `__str__` would just return the value of a field on the model (`self.title`, say), so I think it's clearer to just explicitly include that field in your spec. If you do have some more complex logic in `__str__` then your `display` workaround is fine, but I'm not sure that needs to be right there in the tutorial - I think (hope!) that people can work that out for themselves once they've read the rest of the docs.

Regarding putting the examples at the top of the tutorial: doesn't that rather derail the narrative? I think building up to those examples guides understanding more clearly.